src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,128 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int DEFAULT_CAPACITY = 16;
+    private int threshold = (int) (LOAD_FACTOR * DEFAULT_CAPACITY);
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (size > threshold) {
+            resize();
+        }
+        size++;
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, null);
+            return;
+        }
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                node.value = value;
+                size--;
+                return;
+            }
+            if (node.next == null) {
+                break;
+            }
+            node = node.next;
+        }
+        node.next = new Node<>(key, value, null);
 
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
+    @Override
+    public List<V> getValues() {
+        Node<K, V>[] nodes = getAllNodes();
+        List<V> values = new ArrayList<>();
+        for (Node<K, V> node : nodes) {
+            values.add(node.value);
+        }
+        return values;
+    }
+
+    @Override
+    public List<K> getKeys() {
+        Node<K, V>[] nodes = getAllNodes();
+        List<K> keys = new ArrayList<>();
+        for (Node<K, V> node : nodes) {
+            keys.add(node.key);
+        }
+        return keys;
+    }
```

Comment:
maybe it's better to return Set instead of List, because all keys are unique anyway